java/vector/src/main/java/org/apache/arrow/vector/BigIntVector.java

```diff
@@ -339,6 +339,11 @@ public TransferPair makeTransferPair(ValueVector to) {
     return new TransferImpl((BigIntVector) to);
   }
 
+  @Override
+  public void setEncodedValue(int index, int value) {
```

Comment:
why not do this as long and we can have just a generic setWithPossibleTruncate(int index, long value)? Could be generally useful as this naming/sizing is very dictionary specific.

Reply:
I think that is a good suggestion and I suggested a long as well.  @tianchen92 do you want to open a JIRA/PR to refactor the name/size?

Reply:
Sure.